pkg/node/manager/manager.go

```diff
@@ -402,66 +413,45 @@ func (m *manager) NodeUpdated(n nodeTypes.Node) {
 		}
 
 		prefix := ip.IPToNetPrefix(address.IP)
-		ipAddrStr := prefix.String()
-		_, err := m.ipcache.Upsert(ipAddrStr, tunnelIP, key, nil, ipcache.Identity{
-			ID:     remoteHostIdentity,
-			Source: n.Source,
-		})
-		resource := ipcacheTypes.NewResourceID(ipcacheTypes.ResourceKindNode, "", n.Name)
-		m.upsertIntoIDMD(prefix, remoteHostIdentity, resource)
-
-		// Upsert() will return true if the ipcache entry is owned by
-		// the source of the node update that triggered this node
-		// update (kvstore, k8s, ...) The datapath is only updated if
-		// that source of truth is updated.
-		// The only exception are kube-apiserver entries. In that case,
-		// we still want to inform subscribers about changes in auxiliary
-		// data such as for example the health endpoint.
-		overwriteErr := &ipcache.ErrOverwrite{
-			ExistingSrc: source.KubeAPIServer,
-			NewSrc:      n.Source,
-		}
-		if err != nil && !errors.Is(err, overwriteErr) {
-			dpUpdate = false
-		} else {
-			ipsAdded = append(ipsAdded, ipAddrStr)
+		m.ipcache.UpsertMetadata(prefix, n.Source, resource,
+			nodeLabels,
+			ipcacheTypes.TunnelPeer(tunnelIP),
+			ipcacheTypes.EncryptKey(key))
+		if nodeIdentityOverride {
+			m.ipcache.OverrideIdentity(prefix, nodeLabels, n.Source, resource)
 		}
+
+		nodeIPsAdded = append(nodeIPsAdded, prefix)
 	}
 
 	for _, address := range []net.IP{n.IPv4HealthIP, n.IPv6HealthIP} {
-		if address == nil {
+		healthIP := ip.IPToNetPrefix(address)
+		if !healthIP.IsValid() {
 			continue
 		}
-		addrStr := address.String()
-		_, err := m.ipcache.Upsert(addrStr, nodeIP, n.EncryptionKey, nil, ipcache.Identity{
-			ID:     identity.ReservedIdentityHealth,
-			Source: n.Source,
-		})
-		if err != nil {
-			dpUpdate = false
-		} else {
-			healthIPsAdded = append(healthIPsAdded, addrStr)
-		}
+		m.ipcache.UpsertMetadata(healthIP, n.Source, resource,
```

Comment:
Strictly speaking the bug I'm referring to above is probably not possible until we switch CIDR policy over to the new API, but it'd be nice to add the one extra override line in this PR since it's handling the rest of the related code here, then the future PR won't have to worry about (or forget about) this detail.

Reply:
Discussed offline: health IPs are already excluded in `resolveIdentity()` but we do need to ensure that ingress IPs are also excluded. I've updated https://github.com/cilium/cilium/issues/21142 so we can followup.

Reply:
https://github.com/cilium/cilium/pull/23208#issuecomment-1509337688